src/transformers/models/speech_to_text/modeling_speech_to_text.py

```diff
@@ -153,9 +153,9 @@ def make_weights(self, num_embeddings: int, embedding_dim: int, padding_idx: Opt
             # in forward put the weights on the correct dtype and device of the param
             emb_weights = emb_weights.to(dtype=self.weights.dtype, device=self.weights.device)
 
-        self.weights = nn.Parameter(emb_weights)
+        self.weights = emb_weights
```

Comment:
@patil-suraj - is this change ok? Why exactly do we have to do this here actually?

Reply:
I'm not sure we can just save a tensor like this in PyTorch. When doing `model.to(device)` does this tensor correctly move to the device as well?

Reply:
I don't really understand why this would cause an issue in TF<=>PT exactly - what was the issue? BTW, we always wrap the static sinusoidal encodings in a `nn.Parameter()` or `nn.Embedding(...)`, E.g. Vit's positional embeddings, DistilBERT's position embeddings. Why should we do it differently here?

Reply:
> what was the issue?

This is the only model that I could find that may change the size of the embedding layer in the forward call ([here](https://github.com/huggingface/transformers/blob/master/src/transformers/models/speech_to_text/modeling_speech_to_text.py#L188)). The TF version for other embeddings allocates these named model parameters in [`build()`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#build), which is expected to happen before the first `call()`. Since `nn.Parameter()` is mostly a wrapper to add the variable to the model's list of parameters (AFAIK), then dropping it means we don't need a named parameter in TF, which means we can simplify the code and avoid solving the resizing issue.

Reply:
Technically, someone who fine-tuned (or just saved and reshared) a Speecj2Text model will have this weight in the checkpoint. I don't know how it's used exactly, so I don't know if it's breaking to ignore it (we would have to put it in the `_keys_to_ignore_on_load_unexpected`).

Reply:
(sorted on Slack -- conclusion: reverting this change since it is needed for the `to` method)